lib/internal/url.js

```diff
@@ -1512,32 +1512,32 @@ function fileURLToPath(path, options = kEmptyObject) {
 
 function pathToFileURL(filepath, options = kEmptyObject) {
   const windows = options?.windows ?? isWindows;
-  if (windows && StringPrototypeStartsWith(filepath, '\\\\')) {
+  let resolved = windows ? path.win32.resolve(filepath) : path.posix.resolve(filepath);
```

Comment:
Instead of introducing a new variable named `resolved` and renaming the existing parameter `filepath`, rename the input parameter to `filepathOriginal` and create a new variable `filepath` for the modified version of the original path.

```suggestion
  let filepath = windows ? path.win32.resolve(filepathOriginal) : path.posix.resolve(filepathOriginal);
```

Reply:
I'm not introducing it, it already exists; I'm only moving its declaration upper.